core/src/main/java/org/apache/druid/common/config/NullHandling.java

```diff
@@ -50,9 +49,17 @@
    * It does not take effect in all unit tests since we don't use Guice Injection.
    */
   @Inject
-  private static NullValueHandlingConfig INSTANCE = new NullValueHandlingConfig(
-      Boolean.valueOf(System.getProperty(NULL_HANDLING_CONFIG_STRING, "true"))
-  );
+  private static NullValueHandlingConfig INSTANCE;
```

Comment:
I suggest making a private `getInstance()` method, using that everywhere, and having that method throw a nice exception that asks you to either call `NullHandling.initializeForTests()` or include `NullHandlingModule`. That way, future benchmark and test writers don't get confused by null pointer exceptions.